libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/arrow/RouteArrowUtils.kt

```diff
@@ -60,10 +60,26 @@ internal object RouteArrowUtils {
             return
         }
 
-        val aboveLayerIdToUse = if (style.styleLayerExists(options.aboveLayerId)) {
-            options.aboveLayerId
-        } else {
-            null
+        val aboveLayerIdToUse = when {
+            options.aboveLayerId == RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID -> {
+                when {
+                    style.styleLayerExists(RouteLayerConstants.RESTRICTED_ROAD_LAYER_ID) -> {
+                        RouteLayerConstants.RESTRICTED_ROAD_LAYER_ID
+                    }
+                    style.styleLayerExists(RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID) -> {
+                        RouteLayerConstants.PRIMARY_ROUTE_TRAFFIC_LAYER_ID
+                    }
+                    else -> {
+                        null
+                    }
+                }
+            }
```

Comment:
Is there a more accessible way for us to help developers determine the reference layer ID instead of this type of hack?

Maybe a static utility:
```
fun RouteLayerConstants.getTopRouteLayerId(): String
```
that developers can use when constructing the `RouteArrowOptions` (or anything else that tries to place something on top of the route) that we would keep up-to-date if for some reason the ordering layers becomes different, or a new layer is added? Otherwise nearly each change like this in the future would be a breaking change.

Reply:
I created a top level route line layer with an empty source that will always be the top layer so it takes the place of what was previously the top layer, the traffic layer.  This way we can do what ever we want below that layer and it can always be relied upon to be the top route line related layer.  wdyt?